src/main/java/io/jenkins/plugins/artifact_manager_s3/S3BlobStoreConfig.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.artifact_manager_s3;
+
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang.StringUtils;
+import org.jclouds.aws.domain.Region;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.model.GlobalConfiguration;
+
+/**
+ * Store the S3BlobStore configuration to save it on a separate file. This make that
+ * the change of container does not affected to the Artifact functionality, you could change the container
+ * and it would still work if both container contains the same data.
+ */
+@Extension
```

Comment:
This does not work in `JCloudsVirtualFileTest`. You could pull the `JenkinsRule` field up from `JCloudsArtifactManagerTest` to `S3AbstractTest`; or amend `S3BlobStore.getConfiguration` to create a new empty `S3BlobStoreConfig` (its `container` and `prefix` fields should be ignored anyway due to test overrides).